app/src/main/java/com/example/filemanager/AllFilesFragment.kt

```diff
@@ -65,15 +83,34 @@ class AllFilesFragment : Fragment(), ItemClickListener {
         fetchFiles()
     }
 
-    override fun onItemClickListener() {
-        Toast.makeText(context, "Item clicked", Toast.LENGTH_SHORT).show()
+    private fun setToolbarNavigation() {
+        val toolbar = activity?.findViewById<Toolbar>(R.id.toolbar)
+        (activity as AppCompatActivity?)!!.setSupportActionBar(toolbar)
+        (activity as AppCompatActivity?)!!.supportActionBar?.setDisplayHomeAsUpEnabled(true)
```

Comment:
Please always use `requireActivity()` , also can extract the activity retrieval to a method 

`getMainActivity(): MainActivity = requireActivity() as MainActivity`
then use
`getMainActivity().setSupportActionBar(toolbar)`